plugins/mongodb/mongodbembed.hpp

```diff
@@ -353,39 +383,60 @@ namespace mongodbembed
         }
 
         /**
-         * @brief Configures the MongoDB instance and sets up the MongoDB pool. It 
-         * should only be called once because only one instance object is allowed at
-         * a time.
+         * @brief Configures the MongoDB instance for the client objects to use for connections. It 
+         * should only be called once because only one instance object is allowed per program.
          * 
          * @param instance The instance object that is to be kept alive for multiple
          * threads to have access to.
-         * @param pool The pool object that has the information about which database
-         * and collection to connect to.
          */
-        void configure(std::unique_ptr<mongocxx::instance> && instance, std::unique_ptr<mongocxx::pool> && pool) 
+        void configure(std::unique_ptr<mongocxx::instance> && instance) 
         {
             _instance = std::move(instance);
-            _pool = std::move(pool);
         }
 
-        using connection = mongocxx::pool::entry;
+        /**
+         * @brief Creates a client object using the specified connections string. The client is
+         * added to the map of active connections.
+         * 
+         * @param connectionString The connection string for constructing the client object.
+         * 
+         * @param queryString A const char * holding the query string for hashing.
+         */
+        void create_connection(const char *connectionString, const char * queryString) 
+        {
+            std::shared_ptr<mongocxx::client> client_ptr = std::make_shared<mongocxx::client>(mongocxx::client{mongocxx::uri{connectionString}});
+
+            // Use a hash of the connection string as the key to finding
+            // any connection objects
+            std::string hashString = std::string(connectionString) + std::string(queryString);
```

Comment:
minor:  You can avoid the temporary string by calling hash on each in turn.  (More significant in lookup)

hash64_t key = rtlHash64VStr(queryString, rtlHash64VStr(connectionString,0));

Reply:
Changed